kvdb/sqlite/db.go

```diff
@@ -50,6 +54,18 @@ func NewSqliteBackend(ctx context.Context, cfg *Config, dbPath, fileName,
 			value: "WAL",
 		},
 	}
+
+	// If the database is being created, set auto_vacuum to incremental.
+	// This must be done before any tables are created. According to the
+	// SQLite docs, this corresponds to mode 2.
+	if isCreate {
```

Comment:
do we want to handle the case for DBs that already exist though so that we can vacuum them too? 
Afaiu, this can be done by setting `auto_vacuum` even after the table exists but then calling the `vacuum` command (which will be quite expensive but will only need to happen once) - cause then after that, `incremental_vacuum` can be applied to those DBs too.

Reply:
even if we dont handle that case, is there anything wrong with adding this pragma option even for existing dbs? afaiu, it just wont have an effect at the time of calling `incremental_vacuum` (if we dont call `vacuum` first). 

i also think that if we _do_ at all want to ever handle the case for existing DBs, then now is probably the time to do it since at this point, we know that if `isCreate` is `false`, then we should call `vacuum` - otherwise, we dont need to. Otherwise, another way of dealing with this for future if we dont want to handle it now is just to provide the user the ability to call `vacuum` themselves. 
either way - i think we can just set this pragma regardless

Reply:
> do we want to handle the case for DBs that already exist though so that we can vacuum them too?

IIC from the docs, you can only set this pragma for DBs that are being created for the very first time: 
> Auto-vacuuming is only possible if the database stores some additional information that allows each database page to be traced backwards to its referrer. Therefore, auto-vacuuming must be turned on before any tables are created. It is not possible to enable or disable auto-vacuum after a table has been created.

Reply:
You're correct that the `vacuum` command can be run for any DB. Vacuum is basically similar to the way bolt compaction works. Incremental vacuum only is able to reclaim space at the end of the free list. 

Here's a relevant section of the docs: 
> Note, however, that auto-vacuum only truncates the freelist pages from the file. Auto-vacuum does not defragment the database nor repack individual database pages the way that the [VACUUM](https://www.sqlite.org/lang_vacuum.html) command does. In fact, because it moves pages around within the file, auto-vacuum can actually make fragmentation worse.

Reply:
Simplified the diff quite a bit. We always set the value as it's just a noop when it doesn't actually apply as you alluded to.